pkg/rpcclient/rpc.go

```diff
@@ -974,9 +974,9 @@ func (c *Client) GetRawNotaryPool() (*result.RawNotaryPool, error) {
 }
 
 // GetBlockNotifications returns notifications from a block organized by trigger type.
-func (c *Client) GetBlockNotifications(blockHash util.Uint256, filters ...*neorpc.NotificationFilter) (*result.BlockNotifications, error) {
+func (c *Client) GetBlockNotifications(blockHash util.Uint256, filter *neorpc.NotificationFilter) (*result.BlockNotifications, error) {
 	var resp = &result.BlockNotifications{}
-	if err := c.performRequest("getblocknotifications", []any{blockHash.StringLE(), filters}, resp); err != nil {
+	if err := c.performRequest("getblocknotifications", []any{blockHash.StringLE(), filter}, resp); err != nil {
```

Comment:
You need to check for nil and then create a copy. See how `ReceiveExecutionNotifications` works.

Reply:
And add a test for nil case to `pkg/rpcclient/rpc_test.go`